lib/huginn_acumen_product_agent/acumen_product_agent.rb

```diff
@@ -110,33 +177,66 @@ def handle(event)
-            products = get_product_categories(client, products)
-            products = get_product_contributors(client, products)
-
-            # map attributes
-            products.map do |product|
-                map_attributes(product)
 
-                product['model'].each do |model|
-                    map_attributes(model)
-                end
+            # Load Products
+            fetch_product_bundles(client, ids, digital_formats, ignored_skus)
+        end
 
-                product
-            end
+        private
+
+        # Returns an array of Product objects for the provided product_ids.
+        # Each object is a merged representation of all the individual Acumen tables
+        # that make up a product record with fields mapped to the schema.org/Product
+        # object definition.
+        def fetch_products(acumen_client, product_ids, digital_format_list)
+            products = fetch_inv_product_data(acumen_client, product_ids, digital_format_list)
+            products = fetch_product_marketing(acumen_client, products)
+            products = fetch_product_contributors(acumen_client, products)
+            products = fetch_product_categories(acumen_client, products)
 
             products.each do |product|
-                unless ignore_skus.include?(product['sku'])
-                  create_event payload: product
-                end
+                map_attributes(product)
             end
+
+            return products
         end
 
-        private
+        # Returns an array of product bundles for the provided `products` array.
+        # Each bundle will contain an array of all the product definitions for each
+        # format of a given title.
+        #
+        # NOTE: The generated bundles will contain both active and inactive products
+        # to facilitate product deletion in external systems.
+        def fetch_product_bundles(acumen_client, product_ids, digital_format_list, ignored_skus)
+
+          begin
+            alternate_ids_map = fetch_alternate_format_ids(acumen_client, product_ids)
 
+            bundles = product_ids.map do |id|
+              begin
+                bundle_ids = alternate_ids_map[id]
+                bundle_ids.append(id) unless bundle_ids.include?(id)
+                bundle = fetch_products(acumen_client, bundle_ids.sort, digital_format_list)
+
+                # Filter out any products that are explicitly ignored by SKU
+                bundle.select { |p| !ignored_skus.include?(p['sku']) }
+
+                create_event payload: { products: bundle, status: 200 }
+              rescue => e
+
+              end
+            end
+          rescue AcumenAgentError => e
+            issue_error(e)
+          end
+        end
+
+        # Maps additional Acumen attributes to the `additionalProperty` array
+        # NOTE:  Attributes mapped in this way will be _removed_ from the
+        # `acumenAttributes` array.
         def map_attributes(product)
+
+
```

Comment:
@mrogers-5s This doesn't really affect anything right now but you did mention to me that the additionalProperty field option will have its name changed to format and I assumed when you said that, you meant this will happen within the acumen agent? Will that happen in a separate agent? Will that even happen at all?

Reply:
@Jhalter5Stones That is an option that's configured at runtime.